warp/Network/Wai/Handler/Warp/Header.hs

```diff
@@ -19,27 +19,35 @@ indexRequestHeader hdr = traverseHeader hdr requestMaxIndex requestKeyIndex
 
 idxContentLength,idxTransferEncoding,idxExpect :: Int
 idxConnection,idxRange,idxHost :: Int
-idxContentLength    = 0
```

Comment:
I don't know how performance will change. Does GHC change `fromEnum TransferEncoding` to a constant value when compiling?